core/internal/filetransfer/tasks.go

```diff
@@ -58,6 +58,15 @@ type Task struct {
 
 	// This can be used to cancel the file upload or download if it is no longer needed.
 	Context context.Context
+
+	// Reference to the artifact being transfered
+	Reference *string
+
+	// VersionId is the version of the file we want to download
+	VersionId interface{}
+
+	// Digest is the checksum to ensure the correct files are being downloaded
+	Digest string
```

Comment:
These fields are specific to artifacts and GCS and don't quite belong on a `Task`. Why does `Url` not work?

Reply:
I think we can use `Url` to store the reference string, but we'd still need to pass `VersionId` and `Digest` somewhere in order to download the artifact. What would you suggest for those, if not passing it via `Task`?